scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketIO.java

```diff
@@ -76,6 +77,23 @@ public static <K1, K2> CoGbkWithSecondaryBuilder<K1, K2> read(
     return new CoGbkWithSecondaryBuilder<>(primaryKeyClass, secondaryKeyClass);
   }
 
+  private static void validateInputNameUniqueness(List<BucketedInput<?>> inputs)
+      throws IllegalArgumentException {
+    HashSet<String> inputNames = new HashSet<>();
+    inputs.stream()
+        .forEach(i -> {
+              if (!inputNames.add(i.getTupleTag().getId())) {
```

Comment:
I think I would just change this to:

```
assert(inputNames.add(i.getTupleTag().getId())
```

And not add `IllegalArgumentException` to these signatures

Reply:
oh ok I can do that. Can you please say why that is preferred?

Reply:
Sure! It's just a small enough edge case that I don't think it warrants changing a user-facing API signature for. Additionally it's the convention we're using elsewhere in scio-smb (check out at the constructors for all BucketMetadata implementations, which all use `assert`s)

Reply:
hmm ok I see the signature change is a good reason. But as far as I know the main difference between assert and throwing an exception is the ability to turn assert of in prod (it is a development feature). Isn't that why we use it in other places?

Reply:
in general it is easy to know where assert should be used and where it should not. e.g. when the input is set by user, you should NOT use assertion. In this case "the user" is the developer which chooses the name of the tuple tags? But that is at development time from their POV, while they are using our prod version of lib. confusing!

Reply:
also this is a runtime exception so no need to have it in the signature